packages/astro/src/content/runtime.ts

```diff
@@ -442,6 +445,29 @@ export async function renderEntry(
 		return entry.render();
 	}
 
+	if (entry.isDeferred) {
+		try {
+			// @ts-expect-error	virtual module
+			const { default: contentModules } = await import('astro:content-module-imports');
+			const module = contentModules.get(entry.filePath);
+			const resolvedComponent = await module();
+			
+			return {
+				Content: resolvedComponent.Content,
+				rendered: {
+					metadata: {
+						headings: resolvedComponent.getHeadings?.() ?? [],
+						remarkPluginFrontmatter: resolvedComponent.frontmatter ?? {},
+					},
+				},
```

Comment:
I don't think this needs to be this shape. The `rendered` object is a property of the top-level `entry`, so we don't need to use it here. We should probably just use the same shape as the old implementation, and maybe implement that for markdown too.